cmd/pwp/main.go

```diff
@@ -224,19 +226,21 @@ func extractWordPress(src, dst string) error {
 		if err != nil {
 			return err
 		}
-		defer fileReader.Close()
 
```

Comment:
`defer` is a better pattern to use, but I can see why you are moving it to the bottom.  Defer is "deferred" until the containing function returns. Thus all the files are open...  yup, not ideal. Nice catch!

Instead of removing the `defer` it will be a better pattern to send `file` to a new function where the defer pattern can be used.  This way that function will return for each file ensuring that the file is closed when the function returns.

Reply:
I do love defer, one of the many things I like about go. Which Is why I left it in the other extract functions.
And ye, Nice idea, so send the file to it's own handler function, the it will defer after handing it. Nice! and you can make it generic to reuse with the other extract functions... Nice!

Reply:
I'll go play more and extract the file handling and update this PR, when I get a chance :)